### DISCUSSION TITLE
WebGIS6 - Druckauflösungen

### DISCUSSION THREAD

**Discussion by Andreas-Lampert** [e-netze/webgis-community#198] [Q&A]
Hallo Jürgen!

Ich teste gerade ein paar Druckauflösungen und dabei ist mir bei einer sehr hohen Druckauflösung (300 dpi) ein Problem aufgefallen.

Im Anhang findest du zwei PDFs, 1x ein Druck mit 200 dpi, 1x ein Druck mit 300 dpi. Der Ausschnitt ist exakt gleich, ich habe nur beim Druckwerkzeug die Qualität angepasst. 
Bei allen Druckauflösungen <= 200 dpi sind die Basemap, die Straßen und die Grundstücksgrenzen sichtbar. 
Bei einer Druckauflösung von 300 dpi fallen plötzlich die Grundstücksgrenzen weg. Diese werden am Ausdruck nicht mehr dargestellt. 

[druck_200dpi_webgis-map_06.09.2024_10_43_05.pdf](https://github.com/user-attachments/files/16905337/druck_200dpi_webgis-map_06.09.2024_10_43_05.pdf)
[druck_300dpi_webgis-map_06.09.2024_10_43_34.pdf](https://github.com/user-attachments/files/16905340/druck_300dpi_webgis-map_06.09.2024_10_43_34.pdf)

Der Druck ist im Maßstab 1:10.000, die Maßstabsgrenze für die Darstellung der Grundstücke liegt bei 1:20.000. 
In der API habe ich im Moment testweise einige unterschiedliche Druckqualitäten konfiguriert:
`<add key="qualities-dpi" value="96:Standard,120:Mittel,150:Hoch,200:Mittel sehr hoch,300:Sehr hoch" />`

Wodurch könnte es sein, dass die Grundstücksgrenzen bei einer Druckauflösung von 300 dpi nicht mehr dargestellt werden?
Gibt es Konfig-Optionen, um die Darstellung der Grundstücksgrenzen bei hohen Druckauflösungen anzuzeigen?

Danke!

Lg, Andi

**Comment by gis-eni** [comment]
Hallo Andi,

Gute Frage.

Zuerst einmal meine Fragen:
Ist das bei allen (Vector) Diensten so?
Was sind die Grundstücksgrenzen für ein Dienst? AGS, WMS?

LG Jürgen

**Comment by Andreas-Lampert** [comment]
Hallo Jürgen!

Es handelt sich um AGS-Dienste. Es passiert nicht mit allen AGS-Diensten. Ich habe es mit einer Gewässer-Karte probiert. Du findest das Ergebnis angehängt. 
Bei 200 dpi ist alles ok, bei 300 dpi fallen die Flusskilometer sowie die Beschriftung weg. Betrifft wieder Maßstab 1:10.000.
Im Maßstab 1:5.000 sowie im Maßstab 1:20.000 sind die Ausdrucke ident. 

[300dpi_webgis-map_06.09.2024_11_34_59.pdf](https://github.com/user-attachments/files/16905964/300dpi_webgis-map_06.09.2024_11_34_59.pdf)
[200dpi_webgis-map_06.09.2024_11_34_58.pdf](https://github.com/user-attachments/files/16905965/200dpi_webgis-map_06.09.2024_11_34_58.pdf)

Lg, Andi

**Comment by gis-eni** [comment]
Das ist seltssam. WebGIS übergibt beim Map Request den DPI Wert an einen AGS MapServer. Für mich schaut es es so aus, als ob der dort nicht ankommt oder ignoriert wird.

Bei uns tritt das Problem nicht auf. Seit wann habt ihr das Problem?
Vielleicht ist es eine neue Episode von „Wie interpretiert man einen Maßstab in WebMercator“?

Ich kann leider nicht viel dazu sagen. Das erfordert eine genauere Analyse bei der ich einen Dienst von euch brauche, mit dem ich das nachvollziehen kann.

LG und schönes Wochenende
Jürgen

**Comment by Andreas-Lampert** [comment]
Das Problem trat bei den Tests vom WebGIS6 auf. Ich teste höhere Druckauflösungen und dabei ist es mir aufgefallen. 
Ich werde dir nächste Woche einen Dienst bereitstellen (bin ab MI wieder im Büro). 

Ebenfalls ein schönes Wochenende!

Lg, Andi

**Comment by gis-eni** [comment]
Das Problem habe ich jetzt einmal analysiert:

Hier waren konkret die MaxImageWidth und MaxImageHeight für den AGS Dienst zu klein gewählt. Wird von AGS ein Bild abgeholt, das größer ist, wird kein Fehler zurück gegeben. Hier wird das Bild vom AGS einfach kleiner ausgeliefert. Der geographische Ausschnitt und die Auflösung bleiben dabei gleich. Was sich dabei für das Ergebnis Bild ändert ist der Maßstab (Maßstab, Bildgröße und Auflösung sind von einander abhängig). Darum können bei diesen Request Daten "verschwinden", weil die Maßstabsgrenzen nicht mehr zum gedruckten Bild passen (danke ESRI).

Da das immer wieder einmal passieren kann und man das gar nicht mitbekommt (ESRI liefert ja keinen Fehler), hab ich jetzt folgenden Workaround für AGS MapServer Dienste eingebaut:

* WebGIS überprüft, ob das Bild klein genug ist (die maximale Größe kann von AGS für diesen Dienst abgerufen werden).
Ist das Bild zu groß, wird automatisch die Bildgröße für den Map-Request auf die maximal zulässige Größe verkleinert. Dabei wird auch die Auflösung (DPI) so angepasst, das der Maßstab erhalten bleibt. Im Bild sollten daher alle Layer angezeigt werden, allerdings wird die Qualität (Auflösung) immer schlechter.

* Wir das Bild beim Ausdruck zu groß, bekommt der Anwender einen Hinweis:
![image](https://github.com/user-attachments/assets/115ee594-570b-4f06-a634-9e701f1a6658)
Die gleiche Fehlermeldung ist dann auch im Ausdruck sichtbar. Wenn das auftritt, weiß man als Admin, dass die Maximale Größe im AGS erhöht werden muss.

* Das Problem kann auch schon am Bildschirm auftreten (Monitore mit hoher Auflösung).  Dann wird das Bild zwar geladen und dargestellt (mit richtigem Maßstab). Die Qualität des Bildes wird aber schlechter. Das etwas nicht stimmt, wird das als Fehler im Viewer angezeigt:
![image](https://github.com/user-attachments/assets/c1a1648e-eff7-4370-9259-bc52560b9b99)
Auch dieser Fehler sollte dem Admin sagen, dass die maximale Größe erhöht werden muss.
